### PR TITLE
Rewrite of the govCMS Akamai Fast Purge module

### PR DESCRIPTION
**Pull request by fubarhouse**
## govCMS Akamai Fast Purge 2019!
The akamai module has previously been flagged as an area for improvement and in adopting the new platform this has become a priority with a temporary mitigation already in place. This update will supersede the temporary fix to that issue.

### Changes
* Individual checks for extension types based on category
* Adds a description of enabled extensions
* Adds permissions for access control
* Safely exposes configuration for .edgerc
* Adds configuration checks and getter
* Removes existing textfield
* Adds one-time purge functionality for purging of a single path (relative to `$base_url`)
* Allows for network configuration when communicating withe the API.
* Updates the module to use vertical tabs.
* Changes configuration to use a safe (non-exposed) configuration variable instead of a file
* Adds a simpletest suite
* Adds drush support for simple operations
* Add support for origin url configuration
* Add logging mechanism to opt-in to logging information
* Support for edgerc file additionally to the environment variable
* Manual purge triggers without using the standard `form_submit()` handler function.
* Override for triggering api calls on non-prod environments
* Manual purging is disabled when the environment type isn't production
* Configurable 'section' to target alternative Akamai configuration profiles
* Supports encrypted variables as input to the API in the prescribed format.
* Decryption of the encrypted values can be facilitated by providing the same passcode and encryption type used to create the encrypted data.
* Decryption of the environment variable can be unique on a site-by-site basis depending on the method used to produce the encrypted data.
* Button to purge site-wide cache, controlled by additional permission.
* Setting for an encrypted host path is available which must resolve via DNS to be used. Using encryption is optional however if used must be applied to both the edgerc data and the host path.
* Support for purging of CP codes using the Akamai API.
* Basic DSM message reports for failures and success after purge button submit handlers run.

### Test results
**Tests need to be updated as at 27 February 2019**
```
261 passes, 0 fails, 0 exceptions, and 27 debug messages
```

### CODE REVIEW

File: modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module [hunk 1]
```diff
@@ -12,6 +12,30 @@ use GuzzleHttp\Exception\ClientException;
 
 define('GAFP_WATCHDOG_TYPE', "govcms_akamai_fast_purge");
 
+/**
+ * Implements hook_permission().
+ */
+function govcms_akamai_fast_purge_permission() {
+  return array(
+    'administer edgerc file path' => array(
+      'title' => t('Administer Akaimai\'s configuration file path.'),
```
Comment (tobybellwood) [inline modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module]:
```suggestion
      'title' => t('Administer Akamai configuration file path.'),
```

File: modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module [hunk 2]
```diff
@@ -12,6 +12,30 @@ use GuzzleHttp\Exception\ClientException;
...
+      'description' => t('Access and set the configuration path for Akaimai fast purge.'),
```
Comment (tobybellwood) [inline modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module]:
```suggestion
      'description' => t('Access and set the configuration path for Akamai fast purge.'),
```

File: modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module [hunk 3]
```diff
@@ -12,6 +12,30 @@ use GuzzleHttp\Exception\ClientException;
...
+    ),
+    'administer akaimai configuration' => array(
```
Comment (tobybellwood) [inline modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module]:
```suggestion
    'administer akamai configuration' => array(
```

File: modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module [hunk 4]
```diff
@@ -12,6 +12,30 @@ use GuzzleHttp\Exception\ClientException;
...
+      'title' => t('Administer Akamai fast purge configuration'),
+      'description' => t('Access and set the configuration for Akaimai fast purge to consume.'),
```
Comment (tobybellwood) [inline modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module]:
```suggestion
      'description' => t('Access and set the configuration for Akamai fast purge to consume.'),
```

File: modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module [hunk 5]
```diff
@@ -12,6 +12,30 @@ use GuzzleHttp\Exception\ClientException;
...
+    ),
+    'administer akamai network' => array(
+      'title' => t('Administer Akamai network'),
+      'description' => t('Access and set the configuration for Akaimai network.'),
```
Comment (tobybellwood) [inline modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module]:
```suggestion
      'description' => t('Access and set the configuration for Akamai network.'),
```

File: modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module [hunk 6]
```diff
@@ -23,61 +47,189 @@ function govcms_akamai_fast_purge_menu() {
     'description' => 'Configuration for govCMS Akamai Fast Purge module',
     'page callback' => 'drupal_get_form',
     'page arguments' => ['govcms_akamai_fast_purge_form'],
-    'access arguments' => ['access administration pages'],
+    'access arguments' => ['administer akaimai configuration'],
```
Comment (tobybellwood) [inline modules/govcms_akamai_fast_purge/govcms_akamai_fast_purge.module]:
```suggestion
    'access arguments' => ['administer akamai configuration'],
```

**Comment by fubarhouse** [comment]
Superseded by #127 
Superseded by #128

**Pull request closed without merging**